kong/plugins/hmac-auth/daos.lua

```diff
@@ -17,6 +17,7 @@ return {
       { consumer = { type = "foreign", reference = "consumers", required = true, on_delete = "cascade", }, },
       { username = { type = "string", required = true, unique = true }, },
       { secret = { type = "string", auto = true }, },
+      { public_key = { type = "string" }, },
```

Comment:
how about `typedefs.key { required = false, referenceable = true, encrypted = true }, }, `?
```suggestion
      { public_key = typedefs.key {
          description = "The RSA public key used to validate signature.",
          type = "string",
          encrypted = true, -- Kong Enterprise-exclusive feature, does nothing in Kong CE
          referenceable = true,
      }, },
```

An additional validator to validate it's a RSA key (not EC or ECX keys) would be good.